File: solutions/java/resistor-color-trio/1/src/main/java/ResistorColorTrio.java
import java.util.Arrays;

class ResistorColorTrio {
    String label(String[] colors) {
        int kilo = 1000;
	    int mega = 1000000;
	    int giga = 1000000000;
	    String[] colorlist = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
	    int FirstIndex = Arrays.asList(colorlist).indexOf(colors[0]);
	    int SecondIndex = Arrays.asList(colorlist).indexOf(colors[1]);
	    int ThirdIndex = Arrays.asList(colorlist).indexOf(colors[2]);
	    long ohm = (FirstIndex * 10 + SecondIndex)* (long) Math.pow(10,ThirdIndex);
	    if (ohm >= kilo && ohm < mega) {
	        return ohm / kilo + " kiloohms";
	    } else if (ohm >= mega && ohm < giga) {
	        return ohm / mega + " megaohms";
	    } else if (ohm >= giga) {
            return ohm / giga + " gigaohms";
        } else {
	        return ohm + " ohms";
	    }
	        
        
}
}

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
values = {
    'black': 0,
    'brown': 1,
    'red': 2,
    'orange': 3,
    'yellow': 4,
    'green': 5,
    'blue': 6,
    'violet': 7,
    'grey': 8,
    'white': 9
}

magnitudes = {
    0: "ohms",
    1: "kiloohms",
    2: "megaohms",
    3: "gigaohms"
}

def label(colors):
    value = (values[colors[0]] * 10 + values[colors[1]])
    exponent =  values[colors[2]]
    magnitud = (exponent+1) // 3
    return str(int(value * 10**(exponent - 3*magnitud))) + " " + magnitudes[magnitud]